src/entity_recorder.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use crate::alloc::alloc::{alloc, dealloc, Layout};
+use crate::alloc::vec::Vec;
+use core::any::TypeId;
+use core::ptr::{self, NonNull};
+
+use crate::archetype::TypeInfo;
+use crate::Entity;
+use crate::{align, DynamicBundle};
+
+/// Mechanism for allowing recording of entities for the purpose of spawning them on future
+///
+/// ```
+/// # use hecs::*;
+/// let mut world = World::new();
+/// let reserved_ent = world.reserve_entity();
+/// let mut entity_recorder = EntityRecorder::new();
+/// entity_recorder.record_entity(reserved_ent,(true,0.10));
+/// world.spawn_recorded(&mut entity_recorder); // recorder can now be reused
+/// assert_eq!(world.contains(reserved_ent), true);
+/// ```
+pub struct EntityRecorder {
+    pub(crate) ent: Vec<(Entity, usize)>,
+    storage: NonNull<u8>,
+    layout: Layout,
+    cursor: usize,
+    info: Vec<(TypeInfo, usize, Entity)>,
+    ids: Vec<TypeId>,
+    pub(crate) mark: usize,
```

Comment:
This should be a local variable in the code that's doing the spawning; there's no need for it to exist outside of that.